### ISSUE TITLE
🐛 Panic `byte index 3 is not a char boundary; it is inside '¡' (bytes 2..4) of `\u¡``

### ISSUE THREAD

**Issue by qarmin** [biomejs/biome#4564]
### Environment information

```block
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### What happened?

```
timeout -v 150 biome lint TEST___FILE.ts
```

crashes with
```
/opt/BROKEN_FILES_DIR/grapheme-next-after-containing_IDX_0_RAND_46800340750317345540107_minimized_383.js:1:1 parse ━━━━━━━━━━

  × Expected a statement but instead found '\u¡'.
  
  > 1 │ \u¡
      │ ^^^
  
  i Expected a statement here.
  
  > 1 │ \u¡
      │ ^^^
  

/opt/BROKEN_FILES_DIR/grapheme-next-after-containing_IDX_0_RAND_46800340750317345540107_minimized_383.js:1:2 parse ━━━━━━━━━━

  × invalid digits after unicode escape sequence
  
  > 1 │ \u¡
      │  Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_diagnostics/src/display/frame.rs:240:57
Thread Name: main
Message: byte index 3 is not a char boundary; it is inside '¡' (bytes 2..4) of `\u¡`


##### Automatic Fuzzer note, output status "None", output signal "Some(6)"
```

File - [compressed.zip](https://github.com/user-attachments/files/17796857/compressed.zip)

### Expected result

No crash

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
